app/javascript/app/components/charts/stacked-area/stacked-area-component.jsx

```diff
@@ -226,26 +225,52 @@ class ChartStackedArea extends PureComponent {
               if (point.y === null) {
                 colorPoint = QUANTIFICATION_COLORS.NOT_QUANTIFIABLE;
               }
-              const yearLabel = isActivePoint ? (
+
+              // yearLabel
+              const LENGHT_LIMIT = 30;
+              const isLongLabel = point.label.length > LENGHT_LIMIT;
+              const yearLabel = (
                 <Label
-                  value={`${point.x} - ${point.label}`}
+                  value={`${point.x}${isLongLabel ? '' : `- ${point.label}`}`}
                   position="top"
                   fill="#8f8fa1"
                   stroke="#fff"
                   strokeWidth={isEdgeOrExplorer ? 0 : 8}
                   style={{ paintOrder: 'stroke' }}
                   fontSize="13px"
                   offset={25}
-                  isFront
                 />
-              ) : null;
+              );
+
+              // extraLabelLine - For long labels
+              const MAX_LINE_LENGHT = 20;
+              const LABEL_OFFSET = 10;
+              const DY = 20;
+              const extraLabelLine = (text, offset) => (
+                <Label
+                  key={text}
+                  value={text}
+                  position="insideTop"
+                  fill="#8f8fa1"
+                  stroke="#fff"
+                  strokeWidth={isEdgeOrExplorer ? 0 : 8}
+                  style={{ paintOrder: 'stroke', zIndex: 500 }}
+                  fontSize="13px"
+                  offset={offset}
+                />
+              );
+              const extraLabel = wordWrap(
+                point.label,
+                MAX_LINE_LENGHT
+              ).map((l, i) => extraLabelLine(l, LABEL_OFFSET + i * DY));
```

Comment:
Is kind of cryptic whats happening in the `offset`  parameter, wouldn't it be possible to set a fixed value?   
What`s the aim of using the `index`?